www/src/components/MarkdownPage/MarkdownPage.js

```diff
@@ -21,102 +21,146 @@ import StickyResponsiveSidebar from 'components/StickyResponsiveSidebar';
 import TitleAndMetaTags from 'components/TitleAndMetaTags';
 import findSectionForPath from 'utils/findSectionForPath';
 import toCommaSeparatedList from 'utils/toCommaSeparatedList';
-import {sharedStyles} from 'theme';
+import {media, sharedStyles} from 'theme';
 import {urlRoot} from 'constants';
 
-const MarkdownPage = ({
-  authors,
-  date,
-  ogDescription,
-  location,
-  markdownRemark,
-  sectionList,
-  titlePostfix = '',
-}) => {
-  const hasAuthors = authors.length > 0;
-  const titlePrefix = markdownRemark.frontmatter.title || '';
+class MarkdownPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toggleMenuOverlay = this.toggleMenuOverlay.bind(this);
+    this.handleWindowResize = this.handleWindowResize.bind(this);
+    this.state = {
+      isMenuOverlayOpen: false,
+    };
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.isMenuOverlayOpen && !prevState.isMenuOverlayOpen) {
+      // Menu has been opened
+      window.addEventListener('resize', this.handleWindowResize);
+    }
+  }
+  handleWindowResize() {
+    const mediaQuery = media.greaterThan('small').replace('@media ', '');
+    if (window.matchMedia(mediaQuery).matches) {
+      // If the screen has been resized >= 'small'
+      this.toggleMenuOverlay(false);
+    }
+  }
+  toggleMenuOverlay(willOpen) {
+    this.setState(
+      {
+        isMenuOverlayOpen: willOpen,
+      },
+      () => {
+        this.context.onMenuOverlayToggle(this.state.isMenuOverlayOpen);
+      },
+    );
+  }
+  handleTouchMove(evt) {
+    // Prevent anything else from being touched when the menu overlay is open, to avoid touch scrolling issues
+    if (this.state.isMenuOverlayOpen) {
+      evt.preventDefault();
```

Comment:
This seems good, but it also makes me nervous;
- how can we ensure that this will not catch `touchMove` events from within the menu overlay?
- what if we somehow get into a state where `isMenuOverlayOption` is true but the user can't see the menu overlay? Can we add an invariant somehow to prevent that?

I know these are not great suggestions - will try to come up with something more specific on my second pass at this.

Reply:
I feel ya! But it's unfortunately the only fix I've found for the drastic Safari scrolling bugs.

- It doesn't catch touch events from the menu, because of the corresponding `evt.stopPropagation()` I've added.
- I don't see how the user would see the menu overlay, as that's controlled by the open prop, which is derived from the state itself. Can you elaborate? :)